src/pybind/mgr/dashboard/services/oauth2_sso.py

```diff
@@ -0,0 +1,71 @@
+import errno
+
+from .. import mgr
+
+
+class Oauth2:
+    def __init__(self, realm_name=None):
+        self.realm_name = realm_name
+
+    def to_dict(self):
+        return {
+            'realm_name': self.realm_name
+    }
```

Comment:
As we no longer have to support Python 3.6, >=3.7 brings `dataclasses`:
```suggestion
from dataclasses import dataclass, asdict

@dataclass
class Oauth2:
    realm_name: str
```
If you don't want to use `dataclasses.asdict()` you can hard-wire that as:
```suggestion
from dataclasses import dataclass, asdict

@dataclass
class Oauth2:
    realm_name: str
    to_dict = asdict
```